packages/core/src/InstanceSettings.ts

```diff
@@ -150,7 +150,9 @@ export class InstanceSettings {
 	get isDocker() {
 		try {
 			return (
-				existsSync('/.dockerenv') || readFileSync('/proc/self/cgroup', 'utf8').includes('docker')
+				existsSync('/.dockerenv') ||
+				readFileSync('/proc/self/cgroup', 'utf8').includes('docker') ||
+				readFileSync('/proc/self/mountinfo', 'utf-8').includes('containerd')
```

Comment:
If any of these throws, the rest of the checks are ignored. Could we add separate try catches for each?

Reply:
rewrote the function to handle these errors.